download/buffer.go

```diff
@@ -0,0 +1,172 @@
+package download
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/replicate/pget/optname"
+	"io"
+	"math"
+	"math/rand"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/spf13/viper"
+)
+
+const (
+	retryDelayBaseline = 100 // in milliseconds
+	retrySleepJitter   = 500 // in milliseconds (will add 0-500 additional milliseconds)
+
+	retryMaxBackoffTime = 3000 // in milliseconds, do not backoff further than 3 seconds
+	retryBackoffIncr    = 500  // in milliseconds, backoffFactor^retryNum * backoffIncr
+	retryBackoffFactor  = 2    // Base for POW()
+)
+
+var (
+	fileSize int64
+)
+
```

Comment:
this used to be necessary so that extractTarFile could decompression throughput, but now downloadFileToBuffer is passed as an explicit argument and this isn't needed

Reply:
Yeah, this goes away in a future commit :)